src/connection.ts

```diff
@@ -61,23 +29,19 @@ type Constructor<
  * - if the servers fails to publish a message to the client, the WS connection is destroyed
  */
 export class StableWSConnection<
+  // CAUTION: generics are out of usual order here
```

Comment:
Why can't we have them in the usual order?

Reply:
I see the previous interface with less args was different. 👍

Reply:
yea, but actually I'm thinking we can reorder them correctly, this is a private/internal class and should not be considered a breaking change, @vishalnarkhede wdyt?

Reply:
also wont matter much since we're going to refactor the generics